File: 8.Sorting/Java/1_simple_sort.java
class SimpleSort {
    void siSort(int arr[]) {
        for (int i=0; i<arr.length; i++) {
            for (int j=i+1; j<arr.length; j++) {
                if (arr[i] > arr[j]) {
                    int temp = arr[i];
                    arr[i] = arr[j];
                    arr[j] = temp;
                }
            }
        }
        
    }
    void show(int arr[], String msg) {
        System.out.println(msg);
        for (int i=0; i<arr.length; i++) {
            System.out.print(arr[i]+" ");
        }
        System.out.println();

    }
    public static void main(String args[]) {
        int arr[] = {22, 18, 2, 42, 1, 10, 0};
        SimpleSort o = new SimpleSort();
        // SimpleSort o = new SimpleSort(System.in);
        o.show(arr, "before");
        o.siSort(arr);
        o.show(arr, "after");

    }
}

File: 8.Sorting/GoLang/1_simple_sort.go
package main

import "fmt"

func main() {
	arr := []int{99, 2, 42, 10, 0, 22, 1}
	i := 0
	var temp int
	for i < len(arr) {
		j := i + 1
		for j < len(arr) {
			if arr[i] > arr[j] {
				temp = arr[i]
				arr[i] = arr[j]
				arr[j] = temp
			}
			j++
		}
		i++
	}

	fmt.Printf("after \n %v", arr)
}
